validate.go

```diff
@@ -46,16 +49,30 @@ func ValidateJSONSchema(path string) func(http.Handler) http.Handler {
 				http.Error(w, "Failed to read body: "+err.Error(), http.StatusBadRequest)
 				return
 			}
-			validationResult, err := schema.Validate(gojsonschema.NewStringLoader(string(buf)))
-			if err != nil {
-				http.Error(w, "Failed to validate: "+err.Error(), http.StatusInternalServerError)
-				return
-			}
-			if !validationResult.Valid() {
-				validationResultJSON, _ := json.Marshal(ValidationResult{*validationResult})
-				http.Error(w, string(validationResultJSON), http.StatusBadRequest)
-				return
+			bufReader := bytes.NewReader(buf)
+			dec := json.NewDecoder(bufReader)
+
+			for {
+				var obj interface{}
+				if err := dec.Decode(&obj); err == io.EOF {
+					break
+				} else if err != nil {
+					http.Error(w, "Invalid JSON: "+err.Error(), http.StatusBadRequest)
+				}
+
+				objLoader := gojsonschema.NewGoLoader(obj)
+				validationResult, err := schema.Validate(objLoader)
+				if err != nil {
+					http.Error(w, "Failed to validate: "+err.Error(), http.StatusInternalServerError)
```

Comment:
In this case, obj would have failed to validate, so I'm not really sure what will be in there. We could probably read the same number of bytes or smth and print that out, but it is a bit complicated. It would be a nice addition, but I think this is good enough for now